src/client/contextChannels.ts

```diff
@@ -585,7 +615,25 @@ if (typeof fin !== 'undefined') {
 
         eventHandler.registerDeserializer('window-added', deserializeWindowAddedEvent);
         eventHandler.registerDeserializer('window-removed', deserializeWindowRemovedEvent);
+        eventHandler.registerDeserializer('channel-changed', deserializeChannelChangedEvent);
+
+        rehydrate();
     }, (reason) => {
         console.warn('Unable to register client channel context handlers. getServicePromise() rejected with reason:', reason);
     });
 }
+
+async function rehydrate(): Promise<void> {
```

Comment:
This will likely not play nice with `getOrCreateAppChannel`. Probably an in person conversation

Reply:
This now works with app channels.